### ISSUE TITLE
ProcessResult in InferenceTest.inl is bugged, produces range [0, 1000] or 1001 categories

### ISSUE THREAD

**Issue by petertorelli** [ARM-software/armnn#185]
The code that sorts the predictions from the inference has a bug. While testing 300 images from ILSVRC2012_val (the first 100 + all 50 of category 0 and all 50 of category 999) produce a range of predictions from 0 to 1000, which is 1001 categories, which is not possible.

**Comment by MatthewARM** [comment]
Which test are you running? The TfLite and Tensorflow models are all trained with an extra zeroth class ("background") and therefore there are 1001 activations in the last softmax layer.

**Comment by petertorelli** [comment]
Oh, my mistake.

I'm using the MobileNet_v1_1.0_224 model from here...

https://github.com/tensorflow/models/blob/master/research/slim/nets/mobilenet_v1.md

...and did not realize the output layer is (?, 1001) because I had been working from the ILSVRC2012 validation set only, assuming 50,000 inferences from 1000 classes.

Closing... Thanks!
